File: app/src/main/java/ezike/tobenna/myweather/data/local/LocalDataSource.java
package ezike.tobenna.myweather.data.local;

import ezike.tobenna.myweather.data.source.BaseSource;

public interface LocalDataSource<K, V> extends BaseSource {

    void save(K k);

    boolean hasLocationChanged(K k);

    boolean shouldFetch(K k);

    @Override
    V get();
}

File: app/src/main/java/ezike/tobenna/myweather/data/local/LocalDataSource.kt
package ezike.tobenna.myweather.data.local

import ezike.tobenna.myweather.data.model.WeatherResponse
import kotlinx.coroutines.flow.Flow


interface LocalDataSource {
    suspend fun save(weatherResponse: WeatherResponse?)
    fun hasLocationChanged(weatherResponse: WeatherResponse): Boolean
    fun getWeather(): Flow<WeatherResponse>
}
